src/coreclr/jit/lclvars.cpp

```diff
@@ -261,7 +261,9 @@ void Compiler::lvaInitTypeRef()
 
 #ifdef TARGET_ARM
     compHasSplitParam = varDscInfo.hasSplitParam;
-#endif
+#elif defined(TARGET_RISCV64)
+    compHasSplitParam = varDsc->lvIsSplit;
```

Comment:
This only checks if the last parameter is split. That doesn't seem correct. You should add a case in the main loop over parameters in `lvaInitUserArgs` instead, next to the ARM32 code that sets `InitVarDscInfo::hasSplitParam` (and make that field available on RISCV64 as well).

Reply:
In fact, it checks whether the first non-parameter local is split, which seemingly should always be false..? `varDsc` might also not be a valid pointer here.

Reply:
Thanks!